### ISSUE TITLE
A11y enhancements to crime and firelog embeds

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#6443]
- table captions that change based on the date range or recent display of content.
nid/311 nid/316
Caption, written in js in text area, showing something like "Showing data from this date range"
- drupal.announce polite notifications on button click that the information is changing.
Drupal notif can be "Fetching data from this date range." Possible change focus?
https://github.com/uiowa/uiowa/issues/6443#issuecomment-1559989740
- Summary?
- date picker limits (not future, not back to the beginning of unix time)?
JS set date limit from widgets.
Fix metadata if you can. Ask ben Why its in the body?

**Comment by briand44** [comment]
keep it scoped to these two forms.

**Comment by joewhitsitt** [comment]
docroot/modules/custom/uiowa_hours/src/Form/HoursFilterForm.php is an example where we have custom announce messages

**Comment by GaryRidgway** [comment]
Thinking on this as I am developing, it seems as though we are trying to do all the development on prod with JS. As I am observing the drupal.announce options, it has dawned on me that this will be the only drupal integration that we will have, and might require a site split, release, etc. Is there some way that I could emulate the experience desired within JS instead of through Drupal?

**Comment by bspeare** [comment]
Can we just add `- core/drupal.announce` as a dependency in `uids_base.libraries.yml` or is there more to it?

**Comment by joewhitsitt** [comment]
Good point.

>Can we just add - core/drupal.announce as a dependency in uids_base.libraries.yml

If that doesn't work I think implementing some basic aria-live polite functionality will be sufficient. It won't add it to the drupal.announce region that is available but will still accomplish what we want.

Another example where drupal announce is called in JS: docroot/sites/brand.uiowa.edu/modules/brand_core/js/lockup-preview.behaviors.js

random google search result for implementing aria-live
https://bitsofco.de/using-aria-live/

**Comment by GaryRidgway** [comment]
Okay, I have decided to implement some `aria-live` then. I decided not to change any focus, because after doing a little reading, It seems like focus changing should be more about going in and out of contexts, and `aria-live` should tell the screen reader when html changes anyways.

Last thing, for summaries, Are they necessary, or are captions enough? They would probably make more sense if we had a strong basis of what to add in the summary, but we currently don't have one. Any ideas?

**Comment by joewhitsitt** [comment]
i don't think we have good copy for a table summary. caption in my opinion should be sufficient.

**Comment by GaryRidgway** [comment]
> i don't think we have good copy for a table summary. caption in my opinion should be sufficient.

Sounds good! Thanks @joewhitsitt.

Anyone want to review?
https://safety.prod.drupal.uiowa.edu/crime-log
and 
https://safety.prod.drupal.uiowa.edu/clery-edge-firelog-test

**Comment by GaryRidgway** [comment]
Based on the feedback gotten in the EOD check-in on Mar 31, 2023, It was decided that the captions should change to more relevantly state what Type of data they are displaying. Additionally, aria-live should not contain the whole table as it will read the entirety of the table when it changed. We should put the aria-live only on the caption so that it is descriptive what changed.

These updates have been made.
